docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,110 @@
+All-capital words are interpreted per [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+# Compiler Command Arguments
+
+The compiler *MUST* receive a new switch, `/editorconfig`, which accepts a single
+file which is an .editorconfig file that may apply to one or more files in the
+compilation. Multiple files are passed with multiple `/editorconfig` switches,
+the same way you can pass multiple references with /reference or multiple analyzers
+with /analyzer. There is no short form of the argument. Files passed with relative
+file names are resolved relative to the current working directory the same
+as source files.
+
+Examples:
+
+```
+    csc.exe /editorconfig:.editorconfig
+    csc.exe /editorconfig:.editorconfig /editorconfig:..\.editorconfig
+    csc.exe /editorconfig:.editorconfig /editorconfig:Subdirectory\.editorconfig
+```
+
+It is expected that the compiler is given the .editorconfig files with casing that
+matches the underlying casing of the file system. The compiler *MUST NOT* do any case correction. So
+
+```
+    csc.exe /editorconfig:Subdirectory\.editorconfig /editorconfig:SubDirectory\.editorconfig
+```
+
+is a valid input to the compiler. The compiler *MAY* choose to issue a warning if two paths given
+are exactly equal by ordinal in a case-sensitive way.
+
+If multiple .editorconfigs apply to a given file (the precise definition of
```

Comment:
Basically what we're saying is:

1. If you pass the command line correctly, we'll behave correctly according to EditorConfig
2. If you pass the command line in a different order, we'll apply some other semantics to the EditorConfig, leading to out-of-order evaluation
3. The correct behavior (according to EditorConfig semantics) is obvious in both cases 1 and 2, but we don't feel like doing work to make it actually behave that way

As long as the absolute path to the individual .editorconfig files passed in matters, which we know it does if we plan to allow `/editorconfig:..\.editorconfig` as an argument, I see no reason to not just implement proper path ordering in the compiler. Especially considering how easy it is since the containment logic is already required for evaluating the individual sections.